Common subdirectories: systemtap-2.6.orig/doc and systemtap-2.6/doc
Common subdirectories: systemtap-2.6.orig/emacs and systemtap-2.6/emacs
Common subdirectories: systemtap-2.6.orig/EXAMPLES and systemtap-2.6/EXAMPLES
Common subdirectories: systemtap-2.6.orig/includes and systemtap-2.6/includes
Common subdirectories: systemtap-2.6.orig/initscript and systemtap-2.6/initscript
Common subdirectories: systemtap-2.6.orig/java and systemtap-2.6/java
Common subdirectories: systemtap-2.6.orig/m4 and systemtap-2.6/m4
Common subdirectories: systemtap-2.6.orig/man and systemtap-2.6/man
Common subdirectories: systemtap-2.6.orig/po and systemtap-2.6/po
Common subdirectories: systemtap-2.6.orig/runtime and systemtap-2.6/runtime
Common subdirectories: systemtap-2.6.orig/scripts and systemtap-2.6/scripts
diff -u systemtap-2.6.orig/session.cxx systemtap-2.6/session.cxx
--- systemtap-2.6.orig/session.cxx	2014-09-05 23:33:21.000000000 +0200
+++ systemtap-2.6/session.cxx	2014-09-30 09:15:53.243980701 +0200
@@ -59,6 +59,26 @@
 bool systemtap_session::NSPR_Initialized = false;
 #endif
 
+static string
+kernel_store_path (void)
+{
+  ifstream proc_cmdline ("/proc/cmdline");
+  string variable_name, store_path;
+
+  getline (proc_cmdline, variable_name, '=');
+  if (variable_name == "BOOT_IMAGE")
+    {
+      string boot_image_path;
+      size_t slash_pos;
+
+      getline (proc_cmdline, boot_image_path, ' ');
+      slash_pos = boot_image_path.find_last_of ('/');
+      store_path = boot_image_path.substr (0, slash_pos);
+    }
+
+  return store_path;
+}
+
 systemtap_session::systemtap_session ():
   // NB: pointer members must be manually initialized!
   // NB: don't forget the copy constructor too!
@@ -102,7 +122,7 @@
   (void) uname (& buf);
   kernel_release = string (buf.release);
   release = kernel_release;
-  kernel_build_tree = "/lib/modules/" + kernel_release + "/build";
+  kernel_build_tree = kernel_store_path() + "/lib/modules/" + kernel_release + "/build";
   architecture = machine = normalize_machine(buf.machine);
 
   for (unsigned i=0; i<5; i++) perpass_verbose[i]=0;
Only in systemtap-2.6: session.cxx~
Common subdirectories: systemtap-2.6.orig/stapdyn and systemtap-2.6/stapdyn
Common subdirectories: systemtap-2.6.orig/staprun and systemtap-2.6/staprun
Common subdirectories: systemtap-2.6.orig/tapset and systemtap-2.6/tapset
Common subdirectories: systemtap-2.6.orig/testsuite and systemtap-2.6/testsuite
Common subdirectories: systemtap-2.6.orig/vim and systemtap-2.6/vim
